lnwallet/btcwallet/btcwallet.go

```diff
@@ -923,18 +923,44 @@ func minedTransactionsToDetails(
 			return nil, err
 		}
 
+		// isOurAddress is a map containing the output indices
+		// controlled by the wallet.
+		// Note: We make use of the information in `MyOutputs` provided
+		// by the `wallet.TransactionSummary` structure that holds
+		// information only if the output is controlled by the wallet.
+		isOurAddress := make(map[int]bool, len(tx.MyOutputs))
+		for _, o := range tx.MyOutputs {
+			isOurAddress[int(o.Index)] = true
+		}
+
 		var destAddresses []btcutil.Address
-		for _, txOut := range wireTx.TxOut {
-			_, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
+		var outputDetails []lnwallet.OutputDetail
+		for i, txOut := range wireTx.TxOut {
+			var address btcutil.Address
+			sc, outAddresses, _, err := txscript.ExtractPkScriptAddrs(
 				txOut.PkScript, chainParams,
 			)
-			if err != nil {
-				// Skip any unsupported addresses to prevent
-				// other transactions from not being returned.
-				continue
+			if err == nil {
+				// Add supported addresses to destAddresses.
+				// Note: This is deprecated but kept for backwards
+				//compatibility.
+				destAddresses = append(destAddresses, outAddresses...)
+
+				// Use the supported address only if it encodes
+				// a single valid address corresponding to the
+				// output script.
+				if len(outAddresses) == 1 && sc != txscript.MultiSigTy {
```

Comment:
Ah, I wasn't aware of this edge case. I wonder if we should just make the `Address` field of `OutputDetail` a slice and add all addresses to it in this case. Then we could get rid of the `DestAddresses` field here (and update any call sites that use the `TransactionDetails`)...

And on the RPC level the deprecated addresses slice would just be a list of all of these. Not sure about turning the `lnrpc.OutputDetail.address` into a slice either, seems a bit overkill if it only contains one address in almost any case. We could still return it empty if there's more than one address in the internal struct. Then an RPC user can still parse the PK script. What do you think?

Reply:
Just trying to keep this internal wallet code as "clean" as possible and move any special cases to the RPC server.

Reply:
> Ah, I wasn't aware of this edge case. I wonder if we should just make the Address field of OutputDetail a slice and add all addresses to it in this case. Then we could get rid of the DestAddresses field here (and update any call sites that use the TransactionDetails)...

I will update the code to do that!

> And on the RPC level the deprecated addresses slice would just be a list of all of these. Not sure about turning the lnrpc.OutputDetail.address into a slice either, seems a bit overkill if it only contains one address in almost any case. We could still return it empty if there's more than one address in the internal struct. Then an RPC user can still parse the PK script. What do you think?

I think this would also be acceptable. Now the only way that a MultiSig output script would still be represented by a (wrong) P2PKH address is if it is a 1-of-1 multisig. But I think for such an edge case that is OK, since I don't see how this could be abused as the MultiSig script must be 1-of-1 and cannot contain some additional spending paths, right? 

Additionally, I was thinking about adding the address type as a field to the `OutputDetail` structure, just as for `Utxso` and returned when requesting unspents?

Reply:
> if it is a 1-of-1 multisig. But I think for such an edge case that is OK

I agree.

> I was thinking about adding the address type as a field to the OutputDetail structure

That could be useful, yes. But you're thinking about the type returned from `txscript.ExtractPkScriptAddrs()`, right? For that we'd probably need a new enum on the RPC level because the existing `AddressType` only knows P2WKH and NP2WKH (and also has a slightly different context than what we're talking about here).

If you're willing to add that, please go ahead :) My naming suggestion for the enum would be `OutputScriptType` or something like that.